Tests/tests/test_RDKShell.cpp

```diff
@@ -0,0 +1,1041 @@
+                                                                                             "\"callsign\": \"org.rdk.Netflix\","
+                                                                                             "\"visible\": true}"), response));
+    EXPECT_EQ(response, _T("{\"success\":true}"));
+
+    ON_CALL(compositormock, setVisibility(::testing::_, ::testing::_))
+        .WillByDefault(::testing::Return(false));
+    EXPECT_EQ(Core::ERROR_GENERAL, handler.Invoke(connection, _T("setVisibility"), _T("{\"client\": \"org.rdk.Netflix\","
+                                                                                             "\"callsign\": \"org.rdk.Netflix\","
+                                                                                             "\"visible\": true}"), response));
+    ON_CALL(compositormock, getVisibility(::testing::_, ::testing::_))
+	    .WillByDefault(::testing::Invoke(
+                [](const std::string& client, bool& visible){
+                      bool x = true;
+		      visible = x;
+                      return true;
+                }));
+
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getVisibility"), _T("{\"client\": \"org.rdk.Netflix\","
+                                                                                             "\"callsign\": \"org.rdk.Netflix\"}"), response));
+    EXPECT_EQ(response, string("{\"visible\":true,\"success\":true}"));
+}
+
+TEST_F(RDKShellTest, getSystemMemory)
+{
+    EXPECT_CALL(rdkshellapimock, systemRam(::testing::_, ::testing::_, ::testing::_, ::testing::_))
+	    .Times(1)
+            .WillOnce(::testing::Invoke(
+                [&](uint32_t& freeKb, uint32_t& totalKb, uint32_t& availableKb, uint32_t& usedSwapKb) {
+		freeKb = 994056;
+		totalKb = 2830092;
+		usedSwapKb = 0;
+		availableKb = 764628;
+                return true;
+                }));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getSystemMemory"), _T("{}"), response));
+    EXPECT_EQ(response, string("{"
+                                "\"freeRam\":994056,"
+                                "\"swapRam\":0,"
+                                "\"totalRam\":2830092,"
+				"\"availablememory\":764628,"
+                                "\"success\":true"
+                        "}"));
+}
+
+TEST_F(RDKShellTest, showWaterMark)
+{
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("showWatermark"), _T("{\"show\":true}"), response));
+    EXPECT_EQ(response, _T("{\"success\":true}"));
+    ON_CALL(compositormock, hideWatermark())
+            .WillByDefault(::testing::Invoke(
+                [](){
+                      return true;
+                }));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("showWatermark"), _T("{\"show\":false}"), response));
+    EXPECT_EQ(response, _T("{\"success\":true}"));
+}
```

Comment:
I don't think such test makes sense
how do you verify that "showWatermark" shows/hides the watermark?

Reply:
showWatermark simply set receivedShowWatermarkRequest = true if it is enabled , otherwise it will call hideWatermark() and when RDKShell initializing only, 
RDKShell::Initialize()
{
             if (receivedShowWatermarkRequest)
                  {
                    CompositorController::showWatermark();
                    receivedShowWatermarkRequest = false;
                  }
}

Reply:
you need to test with RDKShell::Initialize() to meaningfully verify it
please don't write tests just to write tests
it is important what behavior the test verifies
return "{\"success\":true}" isn't the only behavior of your API. it does something else as well but it's missing in your test